pkg/imgpkg/cmd/fake_registry_test.go

```diff
@@ -33,11 +34,11 @@ func NewFakeRegistry(t *testing.T) *FakeRegistry {
 
 func (r *FakeRegistry) Build() *imagesetfakes.FakeImagesReaderWriter {
 	fakeRegistry := &imagesetfakes.FakeImagesReaderWriter{}
-	fakeRegistry.GenericCalls(func(reference name.Reference) (descriptor v1.Descriptor, err error) {
+	getDescriptor := func(reference name.Reference, r *FakeRegistry) (v1.Descriptor, error) {
```

Comment:
If i'm understanding this correctly, you are asking if the `getDescriptor` function should be a method on `FakeRegistry`. 

I think the short answer is 'no' since `fakeRegistry` is a counterfeiter object of the `ImagesReaderWriter` interface. (It is generated code, and to change it would involve changing the registry interface)

I'm going to resolve this conversation. If this is an incorrect interpretation, let me know and we can improve this part of the code together.